src/generic/collheaderctrlg.cpp

```diff
@@ -77,17 +77,28 @@ bool wxGenericCollapsibleHeaderCtrl::Create(wxWindow *parent,
 
 wxSize wxGenericCollapsibleHeaderCtrl::DoGetBestClientSize() const
 {
-    wxClientDC dc(const_cast<wxGenericCollapsibleHeaderCtrl*>(this));
-    wxSize btnSize = wxRendererNative::Get().GetCollapseButtonSize(const_cast<wxGenericCollapsibleHeaderCtrl*>(this), dc);
+    wxGenericCollapsibleHeaderCtrl* const
+        self = const_cast<wxGenericCollapsibleHeaderCtrl*>(this);
+
+    // The code here parallels that of OnPaint() -- except without drawing.
+    wxClientDC dc(self);
+
+    wxSize size = wxRendererNative::Get().GetCollapseButtonSize(self, dc);
+
     wxString text;
     wxControl::FindAccelIndex(GetLabel(), &text);
-    wxSize textSize = dc.GetTextExtent(text);
-    // Add some padding if the label is not empty
-    if ( textSize.x > 0 )
-        textSize.x += FromDIP(4);
 
-    return wxSize(btnSize.x + textSize.x,
-        wxMax(textSize.y, btnSize.y));
+    const wxSize textSize = dc.GetTextExtent(text);
+
+    size.x += FromDIP(2) + textSize.x;
+    if ( textSize.y > size.y )
+        size.y = textSize.y;
+
+#ifdef __WXMSW__
+    size.IncBy(1);
```

Comment:
Just curious as I always confuse when to use pixels and when DIPs. The commit comment says the focus rectangle width is 1 DIP but to me it looks as if `textSize` is in pixels and you add one pixel to it here? Is this witdh the value obtained with `GetSystemMetrics(SM_C{X,Y}FOCUSBORDER)` or something else?

Reply:
Oops, yes, you're totally right, I actually planned to do just this, but forgot. Thanks for noticing this, I'll add it now.